### PR TITLE
CC-1339: Fix contrast for logo cloud and make banner responsive

### PR DESCRIPTION
**Pull request by glokta1**
**Checklist**:

- [X] I've thoroughly self-reviewed my changes
- [X] I've added tests for my changes, unless they affect admin-only areas (or are otherwise not worth testing)
- [X] I've verified any visual changes using Percy (add a commit with `[percy]` in the message to trigger)

## Summary by CodeRabbit

- **Style**
  - Updated visual appearance of company logos by changing image class attributes to adjust brightness.
  
- **Bug Fixes**
  - Corrected image import paths for various company logos to ensure proper display.

- **Refactor**
  - Modified the `JoinController` to update image imports and variable assignments for banner images.
  - Enhanced the `join.hbs` template to improve responsiveness by adjusting banner display based on screen size.

### CODE REVIEW

**Comment by glokta1** [comment]
(banner solution is a bit hacky. Let me know if we should change this)

File: app/components/affiliate-link-page/logo-cloud.js [hunk 1]
```diff
@@ -9,15 +9,15 @@ import stripeImage from '/assets/images/company-logos/stripe.png';
...
-import dockerImage from '/assets/images/company-logos/docker-company-logo.svg';
+import dockerImage from '/assets/images/company-logos/docker-dark.svg';
 import metaImage from '/assets/images/company-logos/meta.png';
 import grabImage from '/assets/images/company-logos/grab.png';
 import salesforceImage from '/assets/images/company-logos/salesforce.png';
 import substackImage from '/assets/images/company-logos/substack.png';
-import tencentImage from '/assets/images/company-logos/tencent-white.svg';
-import adobeImage from '/assets/images/company-logos/adobe.png';
+import tencentImage from '/assets/images/company-logos/tencent-dark.svg';
+import adobeImage from '/assets/images/company-logos/adobe-company-logo.svg';
```
Comment (rohitpaulk) [inline app/components/affiliate-link-page/logo-cloud.js:20]: This whole thing seems inconsistent - some logos have `-dark`, some have `-company-logo`, some have `company-logo-dark`. I actually don't think we even need this many versions - could we not add `filter: brightness(0);` (that makes the image black except for transparent areas), and then combine that with `opacity: 0.4`?

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Added note re: logos

**Comment by rohitpaulk** [comment]
Banner looks fine! Since it's SVG doesn't have the blurriness problem + even with scaling it looks fine on both desktop & mobile to me. We might want to add links inside there in the future but this is good enough for now!

**Pull request merged**